ansible/manage-node.yml

```diff
@@ -0,0 +1,64 @@
+---
+- name: Manage epoch nodes
+  hosts: all
+  remote_user: epoch
+
+  vars:
+    epoch_bin: "{{ ansible_env.HOME }}/node//bin/epoch"
+    epoch_action: "{{ cmd | default('ping') }}"
+    epoch_running: false
+
+  tasks:
+    - name: Fail on unsupported command
+      fail:
+        msg: "Invalid command. Supported commands are: ping, start, stop, restart"
+      when:
+        - epoch_action != "ping"
+        - epoch_action != "start"
+        - epoch_action != "stop"
+        - epoch_action != "restart"
+      connection: local
+
+    - name: Epoch binary exists
+      stat:
+        path: "{{ epoch_bin }}"
+      register: epoch
+      changed_when: false
+      failed_when: epoch.stat.exists != True
+
+    - name: Ping epoch daemon
+      command: "{{ epoch_bin }} ping"
+      register: ping
+      failed_when: no
+      changed_when: false
+      when: epoch.stat.exists == True
+
+    - name: Register running epoch daemon
+      set_fact:
+        epoch_running: true
+      when: ping.stdout == "pong"
+
+    - name: Fail if not running
+      fail:
+        msg: "Ping failed"
+      when:
+        - not epoch_running
+        - epoch_action == "ping"
+
+    - name: Start epoch deamon
+      command: "{{ epoch_bin }} start"
+      when:
+        - not epoch_running
+        - epoch_action == "start" or epoch_action == "restart"
+
+    - name: Stop epoch deamon
+      command: "{{ epoch_bin }} stop"
+      when:
+        - epoch_running
+        - epoch_action == "stop"
+
+    - name: Restart epoch deamon
+      command: "{{ epoch_bin }} restart"
+      when:
+        - epoch_running
```

Comment:
So if node not running and user issues restart, node stays down without a warning printed? May I suggest deleting restart so user is forced to run stop first then start?

Reply:
This case is handled by always starting the node in https://github.com/aeternity/infrastructure/pull/17/files/8f8e7a237f65aafc928c22b77b00c32d6b68239d#diff-345c1fc9b8d50aad9657a44fa308e014R52